ccl/backupccl: TestDataDriven_column_families failed

ccl/backupccl.TestDataDriven_column_families [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14023462?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/14023462?buildTab=artifacts#/) on release-23.2.1-rc @ [898cd6a363fd47bb92a03bac216f9bed0f64bc08](https://github.com/cockroachdb/cockroach/commits/898cd6a363fd47bb92a03bac216f9bed0f64bc08):


```
        exec-sql [0 args]
        ALTER TABLE cfs SPLIT AT SELECT a FROM cfs;
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:28:
        set-cluster-setting [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:31:
        set-cluster-setting [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:34:
        set-cluster-setting [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:37:
        exec-sql [0 args]
        BACKUP cfs INTO 'nodelocal://1/foo';
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:41:
        exec-sql [0 args]
        CREATE DATABASE r1;
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:45:
        exec-sql [0 args]
        RESTORE cfs FROM LATEST IN 'nodelocal://1/foo' WITH into_db='r1';
        ----
    datadriven.go:144: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3340/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:49:
         SELECT start_key FROM [SHOW RANGES FROM TABLE orig.cfs]
        output didn't match expected:
        @@ -1,6 +1,6 @@
        -…/<TableMin>
        +<before:/Table/106>
         …/1/0
         …/1/1
         …/1/2
         …/1/3
         
    testutils.go:289: no Invalid Descriptors
    testutils.go:266: Warning: Could not check for invalid descriptors: sql: database is closed
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/95e138e66d69292427dfb9528cf06d04/logTestDataDriven_column_families3841091554
--- FAIL: TestDataDriven_column_families (5.30s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_column_families.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36127

--- a/pkg/ccl/backupccl/testdata/backup-restore/column-families
+++ b/pkg/ccl/backupccl/testdata/backup-restore/column-families
@@ -47,10 +47,11 @@ RESTORE cfs FROM LATEST IN 'nodelocal://1/foo' WITH into_db='r1';
 ----
 
 query-sql
-SELECT start_key FROM [SHOW RANGES FROM TABLE orig.cfs]
+SELECT start_pretty from crdb_internal.ranges
+ WHERE start_key >= crdb_internal.index_span('orig.cfs'::regclass::oid::int, 1)[1]
+   AND start_key <= crdb_internal.index_span('orig.cfs'::regclass::oid::int, 1)[2];
 ----
-…/<TableMin>
-…/1/0
-…/1/1
-…/1/2
-…/1/3
+/Table/109/1/0
+/Table/109/1/1
+/Table/109/1/2
+/Table/109/1/3

